src/NHibernate/IStatelessSession.cs

```diff
@@ -74,6 +74,22 @@ public static void FlushBatcher(this IStatelessSession session)
 		{
 			session.GetSessionImplementation().Flush();
 		}
+
+		/// <summary>
+		/// Cancel execution of the current query.
+		/// </summary>
+		/// <remarks>
+		/// May be called from one thread to stop execution of a query in another thread.
+		/// Use with care!
+		/// </remarks>
+		public static void CancelQuery(this IStatelessSession session)
+		{
+			var implementation = session.GetSessionImplementation();
+			using (implementation.BeginProcess())
+			{
+				implementation.Batcher.CancelLastQuery();
+			}
```

Comment:
Granted, `GetSessionImplementation()` is just `=> this;`, but still, calling twice a function while expecting to always get the same thing should not be done.